src/config/scrivitoContentBrowser.js

```diff
@@ -1,107 +1,152 @@
 import * as Scrivito from "scrivito";
 
 Scrivito.configureContentBrowser({
-  filters: {
+  filters: filterContext => {
+    const validObjClasses = filterContext._validObjClasses;
+    if (validObjClasses) {
+      switch (validObjClasses.length) {
+        case 0:
+          return invalidObjClassFilter();
+        case 1:
+          return objClassFilter(validObjClasses[0]);
+        default:
+          return objClassesFilter(validObjClasses);
+      }
+    }
+
+    return defaultFilter();
+  },
+});
+
+function invalidObjClassFilter() {
+  return {
+    _objClass: {
+      options: {
+        All: {
+          title: "Invalid restrictions",
+          selected: true,
+        },
+      },
+    },
+  };
+}
+
+function objClassFilter(objClass) {
```

Comment:
* [ ] FYI We could name this more to the point, e.g. `singleObjClassFilter`. Maybe same for `objClassesFilter` if we can find a better name here.